test/e2e/network/networking.go

```diff
@@ -366,47 +389,57 @@ var _ = SIGDescribe("Networking", func() {
 		// Slow because we confirm that the nodePort doesn't serve traffic, which requires a period of polling.
 		ginkgo.It("should update nodePort: http [Slow]", func() {
 			config := e2enetwork.NewNetworkingTestConfig(f, e2enetwork.UseHostNetwork)
-			ginkgo.By(fmt.Sprintf("dialing(http) %v (node) --> %v:%v (nodeIP) and getting ALL host endpoints", config.NodeIP, config.NodeIP, config.NodeHTTPPort))
-			err := config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
-			if err != nil {
-				framework.Failf("Error dialing http from node: %v", err)
+			for _, ip := range config.NodeIPs {
+				ginkgo.By(fmt.Sprintf("dialing(http) %v (node) --> %v:%v (nodeIP) and getting ALL host endpoints", config.HostTestContainerPod.Spec.NodeName, ip, config.NodeHTTPPort))
+				err := config.DialFromNode("http", ip, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
+				if err != nil {
+					framework.Failf("Error dialing http from node: %v", err)
+				}
 			}
 			ginkgo.By("Deleting the node port access point")
 			config.DeleteNodePortService()
 
-			ginkgo.By(fmt.Sprintf("dialing(http) %v (node) --> %v:%v (nodeIP) and getting ZERO host endpoints", config.NodeIP, config.NodeIP, config.NodeHTTPPort))
-			err = config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, config.MaxTries, sets.NewString())
-			if err != nil {
-				framework.Failf("Error dialing http from node: %v", err)
+			for _, ip := range config.NodeIPs {
+				ginkgo.By(fmt.Sprintf("dialing(http) %v (node) --> %v:%v (nodeIP) and getting ZERO host endpoints", config.HostTestContainerPod.Spec.NodeName, ip, config.NodeHTTPPort))
+				err := config.DialFromNode("http", ip, config.NodeHTTPPort, config.MaxTries, config.MaxTries, sets.NewString())
+				if err != nil {
+					framework.Failf("Error dialing http from node: %v", err)
+				}
```

Comment:
Maybe spend a minute or two thinking if it we could usefully use a helper function here? There's a lot of duplication...